### PR TITLE
Tool Mode for Visualscript

### LINKED ISSUE #17338: Editor tool in visual script
**Godot version:**
900384a622c7f3094540a43670421469e4ee0b06

**Issue description:**
Could not create visual script that is editor tool.
It could be done in gdscript or c# (http://docs.godotengine.org/en/latest/getting_started/scripting/gdscript/gdscript_basics.html#tool-mode)

### PR DESCRIPTION
**Pull request by swarnimarun**
This works but hasn't been stress tested yet so I am gonna leave it as a Draft PR until I get to properly work on it.
Also at the moment, the scene needs to be reloaded for the tool script toggle change to take effect.

*Bugsquad edit:* Fixes #17338.

### CODE REVIEW

**Comment by swarnimarun** [comment]
> Also at the moment, the scene needs to be reloaded for the tool script toggle change to take effect.

This part needs to be fixed for ScriptServers Globally. Haven't looked into fixing it yet.

https://github.com/godotengine/godot/issues/25275

**Comment by swarnimarun** [comment]
Should close https://github.com/godotengine/godot/issues/17338

**Comment by fire** [comment]
There are some clang format problems.

**Comment by swarnimarun** [comment]
Opening it for review/testing.

File: modules/visual_script/visual_script_editor.cpp [hunk 1]
```diff
@@ -3532,6 +3537,11 @@ VisualScriptEditor::VisualScriptEditor() {
 	left_vb->set_v_size_flags(SIZE_EXPAND_FILL);
 	//left_vb->set_custom_minimum_size(Size2(230, 1) * EDSCALE);
 
+	CheckButton *tool_script_check = memnew(CheckButton);
+	tool_script_check->set_text("Make Tool:");
```
Comment (akien-mga) [inline modules/visual_script/visual_script_editor.cpp]: Should use `TTR()` to be translatable.
Also not sure how it looks like, but maybe the colon is redundant with it being a CheckButton? How about "Tool Script" as text?

**Comment by akien-mga** [comment]
Maybe make the commit log a bit more explicit? From simply "Visual Script Tool" it's not clear that this enables VisualScript to be run in-editor as tool scripts (it could also mean "a tool for Visual Script").

**Review by akien-mga — APPROVED** [review]
Style changes requested, otherwise seems good.

**Comment by Jummit** [comment]
> Maybe make the commit log a bit more explicit? From simply "Visual Script Tool" it's not clear that this enables VisualScript to be run in-editor as tool scripts (it could also mean "a tool for Visual Script").
Not to be nitpicky here, but "a tool for Visual Script" still sounds like it is a tool to do something in Visual Script. I think "tool mode for Visual Scripts" explains it perfectly.

**Comment by akien-mga** [comment]
> Not to be nitpicky here, but "a tool for Visual Script" still sounds like
it is a tool to do something in Visual Script. I think "tool mode for
Visual Scripts" explains it perfectly.

That's what I meant indeed. But your proposal sounds good yes.

File: modules/visual_script/visual_script_editor.cpp (hunk 1 above)
Comment (swarnimarun) [reply modules/visual_script/visual_script_editor.cpp]: makes sense.

**Comment by akien-mga** [comment]
@swarnimarun ping.

File: modules/visual_script/visual_script_editor.cpp (hunk 1 above)
Comment (akien-mga) [reply modules/visual_script/visual_script_editor.cpp]: This is still pending.

Comment (swarnimarun) [reply modules/visual_script/visual_script_editor.cpp]: oops, it's fixed now. :)

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**